www/js/survey/enketo-survey.js

```diff
@@ -6,9 +6,48 @@ angular.module('emission.survey.enketo.launch', [
   'emission.plugin.logger',
 ])
 .factory('EnketoSurveyLaunch', function(
-  $rootScope, $ionicPopup, EnketoSurvey, CommHelper
+  $ionicPopup, EnketoSurvey, CommHelper, $ionicModal
```

Comment:
Do you really need two factories, one for the survey and one for the launch? Seems like it would be simpler to combine them, which would also make it easier to follow the logic flow.

Reply:
atm, one factory is enough

Reply:
on second though, I think we better leave 2 factories
One is a low-level factory that directly interface with the library
Another is a high-level factory that help developer structure and prepare the launch of the survey.